### DISCUSSION TITLE
ZWNJ as HALF_SPACE?

### DISCUSSION THREAD

**Discussion by RHJihan** [florisboard/florisboard#2419] [Q&A]
Why is the code for ZWNJ (Zero Width Non-Joiner) character (8204) defined as HALF_SPACE in the KeyCode class?
https://github.com/florisboard/florisboard/blob/b2a1e82963f0d199651a578715cac3482ead99ca/app/src/main/kotlin/dev/patrickgold/florisboard/ime/text/key/KeyCode.kt#L135

When the ZWNJ character is assigned to a key, the associated label defined with the code does not function; instead, it displays as "half space".
`{ "code": 8204, "label": "NJ" }`

**Comment by patrickgold** [comment]
That's an excellent question, it was introduced by commit 0fb7bbb0340453026e36ec4751476ce6c8a3973c 4 years ago so I can't really say why. Is it in the way of an implementation of some sorts?

**Reply by RHJihan** [reply]
I tried to introduce ZWNJ and ZWJ for bn-BD in https://github.com/florisboard/florisboard/pull/2417 as a key. But for this HALF_SPACE const, I can't assign a label for the ZWNJ key.
